Index: \eslint\lib\rules\indent.js
===================================================================
--- \eslint\lib\rules\indent.js
+++ \eslint\lib\rules\indent.js
@@ -337,9 +337,9 @@
      * @param {number} offset The desired indent level
      * @returns {void}
      */
     setDesiredOffset(token, fromToken, offset) {
-        return this.setDesiredOffsets(token.range, fromToken, offset);
+        return token ? this.setDesiredOffsets(token.range, fromToken, offset) : null; // TMCDOS
     }
 
     /**
      * Sets the desired offset of all tokens in a range
Index: \eslint\lib\rules\template-curly-spacing.js
===================================================================
--- \eslint\lib\rules\template-curly-spacing.js
+++ \eslint\lib\rules\template-curly-spacing.js
@@ -131,9 +131,9 @@
 
         return {
             TemplateElement(node) {
                 const token = sourceCode.getFirstToken(node);
-
+                if(!token) return; // TMCDOS 
                 checkSpacingBefore(token);
                 checkSpacingAfter(token);
             }
         };
