Index: \stylelint-webpack-plugin\dist\utils.js
===================================================================
--- \stylelint-webpack-plugin\dist\utils.js
+++ \stylelint-webpack-plugin\dist\utils.js
@@ -1,5 +1,6 @@
 "use strict";
+const UNESCAPED_GLOB_SYMBOLS_RE = /(\\?)([()*?[\]{|}]|^!|[!+@](?=\())/g; // TMCDOS
 
 Object.defineProperty(exports, "__esModule", {
   value: true
 });
@@ -12,9 +13,9 @@
 
 function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
 
 function parseFiles(files, context) {
-  return (0, _arrify.default)(files).map(file => replaceBackslashes((0, _path.join)(context, '/', file)));
+  return (0, _arrify.default)(files).map(file => /*(0, _path.join)*/(replaceBackslashes(context).replace(UNESCAPED_GLOB_SYMBOLS_RE, '\\$2') + '/' + replaceBackslashes(file))); // TMCDOS - https://github.com/mrmlnc/fast-glob/issues/158
 }
 
 function replaceBackslashes(str) {
   return str.replace(/\\/g, '/');
